src/hotspot/share/opto/divnode.cpp

```diff
@@ -1520,12 +1520,14 @@ Node* ModFNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   if (!can_reshape) {
     return nullptr;
   }
+  if (proj_out_or_null(TypeFunc::Control) == nullptr) { // dead node
+    return nullptr;
+  }
+
   PhaseIterGVN* igvn = phase->is_IterGVN();
 
-  bool result_is_unused = proj_out_or_null(TypeFunc::Parms) == nullptr;
-  bool not_dead = proj_out_or_null(TypeFunc::Control) != nullptr;
-  if (result_is_unused && not_dead) {
-    return replace_with_con(igvn, TypeF::make(0.));
+  if (is_unused()) {
+    return make_tuple_of_input_state_and_top_return_values(igvn->C);
```

Comment:
What does this do? Add a short comment :)

Reply:
Same as above: better commented on definition than bloating every single callsite of a function. If I wonder, I go see the definition, if I don't, I don't have useless stuff on my screen.